### PR TITLE
Add random dog image widget example project

### PR DESCRIPTION
**Pull request by tdimnet**
Turn the Random Dog Image hackathon project into an official example:

- Add set up scripts that create a dashboard and app.
- Detail the necessary environment variables

---

By the way, I haven't added the linter rules. If the PR fails to pass the CI, I'll add them today.
EDIT: Yes, it fails. So I'll update the code today to follow the rules 👍

### CODE REVIEW

**Review by joneshf-dd — APPROVED** [review]
This looks great!

As mentioned in Slack, if you want the CI to be green before merging, you shoulc be able to pull the `master` branch (e.g. `git fetch && git merge origin/master && git push`). If that's not really something you want to do, you can merge the PR and it _should_ be fine. If not, we can always address it after the fact.

File: examples/random-dog/dog-api/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const port = process.env.PORT || 3001;
+
+// app.use(express.json())
+app.use(cors());
+
+app.get('/image', async (req, res) => {
+    const breedId = req.query.breed_id;
+    const urlBase = "https://api.thedogapi.com/v1/images/search";
+    const url = breedId == 0 ? urlBase : urlBase.concat(`?breed_id=${breedId}`);
+    console.log(url);
```
Comment (duncanpharvey) [inline examples/random-dog/dog-api/index.js]: Remove unnecessary logging
```suggestion
```

File: examples/random-dog/dog-api/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const imageUrl = await axios.get(url, {
+        headers: {
+            "x-api-key": process.env.API_KEY
+        }
+    }).then(response => {
+        if (response.data.length == 0) return "https://upload.wikimedia.org/wikipedia/en/thumb/7/7e/Datadog_logo.svg/1200px-Datadog_logo.svg.png";
+        return response.data[0].url;
+    });
+    console.log(`imageUrl: ${imageUrl}`);
```
Comment (duncanpharvey) [inline examples/random-dog/dog-api/index.js]: Remove unnecessary logging
```suggestion
```

File: examples/random-dog/dog-image-widget/src/widget/index.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                const options = breeds.map(breed => ({
+                    label: breed['name'],
+                    value: breed['id']
+                }))
+
+                client.dashboard.customWidget.updateOptions([
+                    {
+                        type: WidgetOptionItemType.STRING,
+                        name: 'breed',
+                        label: 'Select a Dog Breed so see Random Images of',
```
Comment (duncanpharvey) [inline examples/random-dog/dog-image-widget/src/widget/index.tsx]: Update label
```suggestion
                        label: 'Select a Dog Breed to get Random Images',
```

File: examples/random-dog/dog-image-widget/src/widget/index.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                .then(({ breeds }) => setBreeds(breeds))
+                .catch(err => console.log('An error occurs on fetching breeds', err))
+    }, [])
+
+    useEffect(() => {
+            if (breeds.length) {
+                const options = breeds.map(breed => ({
+                    label: breed['name'],
+                    value: breed['id']
+                }))
```
Comment (duncanpharvey) [inline examples/random-dog/dog-image-widget/src/widget/index.tsx]: Add `All Breeds` option to widget (already supported by api)
```suggestion
                const options = [{ label: "All Breeds", value: "0" }].concat(breeds.map(breed => ({
                    label: breed['name'] as string,
                    value: breed['id'] as string
                })))
```

File: examples/random-dog/dog-image-widget/src/widget/widget.css [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    border: none;
+    color: white;
+    left: 20px;
+    position: absolute;
+    top: 20px;
+}
+
+.dog-img {
+    object-fit: cover;
+}
```
Comment (duncanpharvey) [inline examples/random-dog/dog-image-widget/src/widget/widget.css:21]: Update css so image covers entire widget area

```suggestion
.main-wrapper {
    height: 100%;
    left: 0px;
    position: absolute;
    top: 0px;
    width: 100%;
}

.image-wrapper {
    height: 100%;
    width: 100%;
}

.new-dog-btn {
    background-color: rgba(100, 44, 165);
    border: none;
    color: white;
    left: 20px;
    position: absolute;
    top: 20px;
}

.dog-img {
    height: 100%;
    object-fit: cover;
    width: 100%;
}
```

Comment (duncanpharvey) [inline examples/random-dog/dog-image-widget/src/widget/widget.css]: Update css so image covers entire widget area

```suggestion
.main-wrapper {
    height: 100%;
    left: 0px;
    position: absolute;
    top: 0px;
    width: 100%;
}

.image-wrapper {
    height: 100%;
    width: 100%;
}

.new-dog-btn {
    background-color: rgba(100, 44, 165);
    border: none;
    color: white;
    left: 20px;
    position: absolute;
    top: 20px;
}

.dog-img {
    height: 100%;
    object-fit: cover;
    width: 100%;
}
```

**Review by duncanpharvey — CHANGES_REQUESTED** [review]
I made a few suggestions for what can be updated in the code. What do you think?

Also two other minor improvements that could be made:
- The New Dog button stays gray after clicking on it. Why does it not change back to the original color after clicking on it?
- In the widget, the image looks better with `object-fit:cover` but in the widget editor the image looks better with `object-fit:contain`. Is there an easy method to add conditional logic to use one css rule in the widget and a different rule in the widget editor?

**Review by duncanpharvey — APPROVED** [review]

**Comment by duncanpharvey** [comment]
Looks great! Thanks for making those updates

**Pull request merged**